include/armadillo_bits/diskio_meat.hpp

```diff
@@ -1269,8 +1269,19 @@ diskio::save_hdf5_binary(const Mat<eT>& x, const hdf5_name& spec)
     
     const std::string tmp_name = diskio::gen_tmp_name(spec.filename);
     
+    //Check if file exists and is HDF5
+    int file_is_hdf5 = arma_H5Fis_hdf5(spec.filename.c_str());
+
     // Set up the file according to HDF5's preferences
-    hid_t file = arma_H5Fcreate(tmp_name.c_str(), H5F_ACC_TRUNC, H5P_DEFAULT, H5P_DEFAULT);
+    hid_t file;
+    if ((file_is_hdf5 == 1) & (spec.append == true))
```

Comment:
The single `&` means bitwise and operation, which is not correct in this instance.  It should be `&&` to indicate logical and operation.